tests/liteclientHSM_test.cc

```diff
@@ -0,0 +1,1179 @@
+    return executeCmd("ostree", {"commit", "--repo", path_, "--branch", branch, "--tree=dir=" + src_dir},
+                      "commit from " + src_dir + " to " + path_);
+  }
+
+  void set_mode(const std::string& mode) {
+    executeCmd("ostree", {"config", "--repo", path_, "set", "core.mode", mode}, "set mode for repo " + path_);
+  }
+
+  const std::string& getPath() const { return path_; }
+
+ private:
+  const std::string path_;
+};
+
+/**
+ * Class SysOSTreeRepoMock
+ *
+ */
+class SysOSTreeRepoMock {
+ public:
+  SysOSTreeRepoMock(std::string _path, std::string _os) : path_{_path}, os_{_os}, repo_{path_ + "/ostree/repo"} {
+    boost::filesystem::create_directories(path_);
+    executeCmd("ostree", {"admin", "init-fs", path_}, "init a system rootfs at " + path_);
+    executeCmd("ostree", {"admin", "--sysroot=" + path_, "os-init", os_}, "init OS in a system rootfs at " + path_);
+    repo_.set_mode("bare-user-only");
+    LOG_INFO << "System ostree-based repo has been initialized at " << path_;
+  }
+
+  const std::string& getPath() const { return path_; }
+  OSTreeRepoMock& getRepo() { return repo_; }
+
+  void deploy(const std::string& hash) {
+    executeCmd("ostree", {"admin", "--sysroot=" + path_, "deploy", "--os=" + os_, hash}, "deploy " + hash);
+  }
+
+ private:
+  const std::string path_;
+  const std::string os_;
+  OSTreeRepoMock repo_;
+};
+
+/**
+ * Class TufRepoMock
+ *
+ */
+class TufRepoMock {
+ public:
+  TufRepoMock(const boost::filesystem::path& _root, std::string expires = "",
+              std::string correlation_id = "corellatio-id")
+      : root_{_root.string()}, repo_{_root, expires, correlation_id}, latest_{Uptane::Target::Unknown()} {
+    repo_.generateRepo(KeyType::kED25519);
+  }
+
+ public:
+  const std::string& getPath() const { return root_; }
+  const Uptane::Target& getLatest() const { return latest_; }
+
+  Uptane::Target addTarget(const std::string& name, const std::string& hash, const std::string& hardware_id,
+                           const std::string& version, const Json::Value& apps_json = Json::Value()) {
+    Delegation null_delegation{};
+    Hash hash_obj{Hash::Type::kSha256, hash};
+
+    Json::Value custom_json;
+    custom_json["targetFormat"] = "OSTREE";
+    custom_json["version"] = version;
+    custom_json[Target::ComposeAppField] = apps_json;
+    repo_.addCustomImage(name, hash_obj, 0, hardware_id, "", null_delegation, custom_json);
+
+    Json::Value target_json;
+    target_json["length"] = 0;
+    target_json["hashes"]["sha256"] = hash;
+    target_json["custom"] = custom_json;
+    latest_ = Uptane::Target(name, target_json);
+    return latest_;
+  }
+
+ private:
+  const std::string root_;
+  ImageRepo repo_;
+  Uptane::Target latest_;
+};
+
+/**
+ * Class DeviceGatewayMock
+ *
+ */
+class DeviceGatewayMock {
+ public:
+  static std::string RunCmd;
+
+ public:
+  DeviceGatewayMock(const OSTreeRepoMock& ostree, const TufRepoMock& tuf, std::string certDir)
```

Comment:
It makes sense to make certDir optional if we would like to reuse implementation.